File: in_java/Pattern_20.java
package in_java;

public class Pattern_20 extends Pattern {
    public Pattern_20(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        byte firstSymbol = 65;
        byte secondSymbol = 1;
        int halfColumn = this.column / 2;

        for (int i = 0; i < this.row; i++) {
            firstSymbol = 65;
            secondSymbol = 1;
            for (int j = 0; j < this.column; j++) {
                if (j >= halfColumn - 1 - i && j <= halfColumn + i) {
                    if (j < halfColumn) {
                        System.out.print((char) firstSymbol);
                        firstSymbol++;
                    } else {
                        System.out.print(secondSymbol);
                        secondSymbol++;
                    }
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_20.rs
pub fn pattern_20() {
    let row = 4;
    let column = 8;
    let mut first_symbol;
    let mut second_symbol;
    let half_column = column / 2;

    for i in 0..row {
        first_symbol = 'A';
        second_symbol = 1;
        for j in 0..column {
            if j >= half_column - 1 - i && j <= half_column + i {
                if j < half_column {
                    print!("{}", first_symbol);
                    let s = first_symbol as u8 + 1;
                    first_symbol = s as char;
                } else {
                    print!("{}", second_symbol);
                    second_symbol += 1;
                }
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
